metrics/prometheus.py

```diff
@@ -15,13 +16,16 @@
 
 
 class Prometheus(object):
-    def __init__(self, url: str, host: str = None):
+    def __init__(self, url: str, host: str = None, pid = 0):
+        self.pid = pid
         self.url = url
-
         self.headers = {}
         if host is not None:
             self.headers['Host'] = host
 
+    def __del__(self):
+        os.kill(self.pid, signal.SIGKILL)
```

Comment:
Not sure I understand this, are we killing pid 0?

Reply:
actually seems like killing pid 0 still working. :) changing to subprocess' pid

the problem without killing will be that port-forwarding still occupies the port, and makes the program unable to rerun.

Reply:
Can you run ./metrics/check_metrics.py and make sure it still works with this, because prometheus forward is killed in `tearDownClass` there.